File: Principles_Of_Programming_Languages/prolog/multiplication_table.pl
%Multiplication_Table_Program

print_multiplication_table(_, 13) :- !.
print_multiplication_table(Number, I) :-  
    format('~w x ~w = ~w~n', [Number, I, Number * I]),  
    NextI is I + 1,  
    print_multiplication_table(Number, NextI).

% Output example:
% ?- print_multiplication_table(5, 1), nl.
% 
% 5 x 1 = 5
% 5 x 2 = 10
% ...
% 5 x 12 = 60

File: Principles_Of_Programming_Languages/pascal/multiplication_table.pas
program multiplication_Table;
(*show the multiplication table of the entered number by user*)

var
  number : integer;
  i : integer;

begin
  writeln('Enter a number: ');
  readln(number);

  for i := 1 to 12 do
    writeln(i, ' x ', number, ' = ', i * number);
end.


(*
output example:

Enter a number:
5
1 x 5 = 5
2 x 5 = 10
3 x 5 = 15
4 x 5 = 20
5 x 5 = 25
6 x 5 = 30
7 x 5 = 35
8 x 5 = 40
9 x 5 = 45
10 x 5 = 50
11 x 5 = 55
12 x 5 = 60
*)